### PR TITLE
handling non-200 responses from all pyxis calls

### PR DESCRIPTION
**Pull request by acornett21**
- Fixes: #415
- Added in checks for non-200 status codes so we do not continue if a pyxis call fails for other non error cases.

This is just a starting point, I'm open to other ideas on how to handle this and maybe the verbiage we want to show the user. Also, the error message we log might need to be formatted better.

**note: only tested with the get call, since we haven't setup a project in pyxis yet**

Signed-off-by: Adam D. Cornett <adc@redhat.com>

### CODE REVIEW

File: certification/errors/errors.go [hunk 1]
```diff
@@ -45,3 +45,4 @@ var ErrInvalidCertProjectId = errors.New("invalid certification project id")
 var ErrNoPyxisAPIKey = errors.New("no pyxis-api-token flag could be found")
 var ErrNoDockerConfig = errors.New("no docker-config flag could be found")
 var ErrSaveFileFailed = errors.New("failed to save file to artifacts directory")
+var ErrNon200StatusCode = errors.New("error while calling red hat submission api")
```
Comment (bcrochet) [inline certification/errors/errors.go:48]: Branding should always be Red Hat.

**Review by bcrochet — CHANGES_REQUESTED** [review]
I will look more closely tomorrow.

File: certification/pyxis/pyxis.go [hunk 2]
```diff
@@ -163,6 +195,13 @@ func (p *pyxisEngine) updateProject(ctx context.Context, certProject *CertProjec
 		return nil, err
 	}
 
+	// statusOk is indicated with 2xx status codes:
+	statusOk := resp.StatusCode >= 200 && resp.StatusCode < 300
```
[the remark this answers is not in the document]
Comment (acornett21) [reply certification/pyxis/pyxis.go]: I refactored this to be a method.

**Review by bcrochet — APPROVED** [review]
/lgtm

**Review by jomkz — APPROVED** [review]

**Pull request merged**